ci/Jenkinsfiles/lib.groovy

```diff
@@ -0,0 +1,148 @@
+  withEnv(["NAMESPACE=${namespace}"]) {
+    sh """
+      ${helmfileCommand} deps
+      ${helmfileCommand} --namespace=${namespace} --environment ${environment} sync
+    """
+  }
+}
+
+void helmfileDestroy(namespace, environment) {
+  def helmfileCommand = getHelmfileCommand()
+  withEnv(["NAMESPACE=${namespace}"]) {
+    sh """
+      ${helmfileCommand} --namespace=${namespace} --environment ${environment} destroy
+    """
+  }
+}
+
+def runFunctionalTests() {
+  def testNuxeoVersions = [
+    '10.10',
+    '2021'
+  ]
+  def stages = [:]
+  for (nuxeoVersion in testNuxeoVersions) {
+    stages["Against Nuxeo ${nuxeoVersion}"] = buildFunctionalTestStage(nuxeoVersion);
+  }
+  parallel stages
+}
+
+def buildFunctionalTestStage(nuxeoVersion) {
+  def environment = "functional-tests-${nuxeoVersion}"
+  def testNamespace = "$CURRENT_NAMESPACE-nuxeo-java-client-ftests-$BRANCH_NAME-$BUILD_NUMBER-${nuxeoVersion.replace('.', '-')}".toLowerCase()
+
+  def ftestsVersion = "${nuxeoVersion}-${VERSION}"
+  return {
+    stage("Against Nuxeo ${nuxeoVersion}") {
+      container("maven") {
+        script {
+          try {
+            // TODO for 10.10 retrieve the last HF version
+            def nuxeoDockerVersion =  nuxeoVersion
+            if ("${nuxeoDockerVersion}"  == "10.10" ) {
+              nuxeoDockerVersion = '10.10-HF56'
+            }
+            echo """
+              ----------------------------------------
+              Build Nuxeo functional tests image
+              ----------------------------------------
+              Nuxeo version: ${nuxeoDockerVersion}
+              Image tag: ${ftestsVersion}
+              Registry: ${DOCKER_REGISTRY}
+            """
+            // push images to the Jenkins internal Docker registry
+            withEnv(["FTESTS_VERSION=${ftestsVersion}", "NUXEO_VERSION=${nuxeoDockerVersion}"]) {
+              sh '''
+                envsubst < ci/docker/nuxeo/skaffold.yaml > skaffold.yaml~gen
+                skaffold build -f skaffold.yaml~gen
+              '''
+            }
+            echo "Create ftests namespace for ${nuxeoVersion}"
+            sh "kubectl create namespace ${testNamespace}"
+
+            echo "Copy image pull secret to ${testNamespace} namespace"
+            sh "kubectl --namespace=platform get secret kubernetes-docker-cfg -ojsonpath='{.data.\\.dockerconfigjson}' | base64 --decode > /tmp/config.json"
+            sh """kubectl create secret generic kubernetes-docker-cfg \
+                --namespace=${testNamespace} \
+                --from-file=.dockerconfigjson=/tmp/config.json \
+                --type=kubernetes.io/dockerconfigjson --dry-run -o yaml | kubectl apply -f -"""
+
+            echo """
+              ----------------------------------------
+              Deploy Nuxeo
+              ---------------------------------------- 
+            """
+            sh "yq write -i ci/helm/environments/${environment}.yaml nuxeo.image.tag ${ftestsVersion}"
```

Comment:
I ended by putting the env directive into the helmfile.yaml directly because it didn't work in the environment values files. This also allows to get rid of these files for a small amount of lines.